src/axom/core/tests/core_benchmark_array.cpp

```diff
@@ -0,0 +1,357 @@
+// Copyright (c) 2017-2024, Lawrence Livermore National Security, LLC and
+// other Axom Project Developers. See the top-level LICENSE file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+#include "benchmark/benchmark.h"
+#include "axom/config.hpp"
+#include "axom/core.hpp"
+#include "axom/slic.hpp"
+#include "axom/fmt.hpp"
+
+#include <tuple>
+#include <vector>
+#include <type_traits>
+#include <cstdint>
+
+//-----------------------------------------------------------------------------
+// Arguments and types to test
+//-----------------------------------------------------------------------------
+
+void CustomArgs(benchmark::internal::Benchmark* b)
+{
```

Comment:
Instead of commenting out, would it better to guard these with a CMake variable, such as 'AXOM_ENABLE_ALL_ARRAY_BENCHMARK_TYPES' which would be false by default?

Reply:
I was thinking about adding a command line arg to enable/disable some tests

Reply:
I added command line args for the array sizes and the features to test in anticipation of significantly more benchmarking of array vs vector in follow-up PRs